_posts/2018-08-20-grpc_on_http2.md

```diff
@@ -0,0 +1,76 @@
+<!--more-->
+
+## gRPC Semantics
+
+To begin, let’s dive into how gRPC concepts relate to HTTP/2 concepts. gRPC introduces three new concepts: *channels* [1], *remote procedure calls* (RPCs), and *messages*. The relationship between the three is simple: each channel may have many RPCs while each RPC may have many messages.
+
+<img src="https://grpc.io/img/channels_mapping_2.png" alt="Channel Mapping" style="max-width: 800px">
+
+Let’s take a look at how gRPC semantics relate to HTTP/2:
+
+<img src="https://grpc.io/img/grpc_on_http2_mapping_2.png" alt="gRPC on HTTP/2" style="max-width: 800px">
+
+Channels are a key concept in gRPC. Streams in HTTP/2 enable multiple concurrent conversations on a single connection; channels extend this concept by enabling multiple streams over multiple concurrent connections. On the surface, channels provide an easy interface for users to send messages into; underneath the hood, though, an incredible amount of engineering goes into keeping these connections alive, healthy, and utilized.
+
+Channels represent virtual connections to an endpoint, which in reality may be backed by many HTTP/2 connections. RPCs are associated with a connection (this association is described further on). RPCs are in practice plain HTTP/2 streams. Messages are associated with RPCs and get sent as HTTP/2 data frames. To be more specific, messages are _layered_ on top of data frames. A data frame may have many gRPC messages, or if a gRPC message is quite large [2] it might span multiple data frames.
+
+## Resolvers and Load Balancers
+
+In order to keep connections alive, healthy, and utilized, gRPC utilizes a number of components, foremost among them *name resolvers* and *load balancers*. The resolver turns names into addresses and then hands these addresses to the load balancer. The load balancer is in charge of creating connections from these addresses and load balancing RPCs between connections.
+
+<img src="https://grpc.io/img/dns_to_load_balancer_mapping_3.png" alt="Resolvers and Load Balancers" style="max-width: 800px">
+
+<img src="https://grpc.io/img/load_balance_round_robins_2.png" alt="Round Robin Load Balancer" style="max-width: 800px">
+
+A DNS resolver, for example, might resolve some host name to 13 IP addresses, and then a RoundRobin balancer might create 13 connections - one to each address - and round robin RPCs across each connection. A simpler balancer might simply create a connection to the first address. Alternatively, a user who wants multiple connections but knows that the host name will only resolve to one address might have their balancer create connections against each address 10 times to ensure that multiple connections are used.
```

Comment:
13 is an unrealistically high number.  In practice it's more like 4 or 1.  Consider making your examples as close to real life as possible, since it makes it easier for readers to identify with them.

Reply:
Ack. The point here is to exaggerate as a form of illustration.